crates/store/src/db/models/queries/transactions.rs

```diff
@@ -126,14 +173,129 @@ pub struct TransactionSummaryRowInsert {
     transaction_id: Vec<u8>,
     account_id: Vec<u8>,
     block_num: i64,
+    initial_state_commitment: Vec<u8>,
+    final_state_commitment: Vec<u8>,
+    input_notes: Vec<u8>,
+    output_notes: Vec<u8>,
+    size_in_bytes: i64,
 }
 
 impl TransactionSummaryRowInsert {
-    fn new(transaction_id: TransactionId, account_id: AccountId, block_num: BlockNumber) -> Self {
+    fn new(
+        transaction_header: &miden_objects::transaction::TransactionHeader,
+        block_num: BlockNumber,
+    ) -> Self {
+        use miden_lib::utils::Serializable;
+
+        // Serialize input notes using binary format (store nullifiers)
+        let input_notes_binary = transaction_header.input_notes().to_bytes();
+
+        // Serialize output notes using binary format (store note IDs)
+        let output_notes_binary = transaction_header.output_notes().to_bytes();
+
+        // Store the size required to store the row + the size that the output notes will take when
+        // casted to NoteSyncRecord (~500B per output note).
+        let size_in_bytes = (transaction_header.to_bytes().len()
+            + (transaction_header.output_notes().len() * 500)) as i64;
```

Comment:
Converting `transaction_header` here to bytes seems a bit wasteful (and I don't think the compiler will be able to optimize this way). Unfortunately, we don't have the `get_size_hint()` function implemented on the `TransactionHeader`. What we could do is manually estimate the size of the header which is:

- 4 bytes for block number.
- 32 bytes for transaction ID.
- 16 bytes for account ID.
- 32 + 32 bytes for initial + final commitments.
- 32 * `num_input_notes` for the input notes.
- 500 * `num_output_notes` for the output notes.

As I mentioned, 500 bytes is a bit of an over-estimate, but it is fine for now. We should explain all of this in the doc comments though.